Subgraph Refactor

0xdavinchee opened the pull request.

0xdavinchee committed:
make events immutable + update some packages

0xdavinchee committed:
Merge branch 'dev' into subgraph-refactor

0xdavinchee committed:
ID! to Bytes!

0xdavinchee committed:
Merge branch 'subgraph-refactor' of github.com:superfluid-finance/protocol-monorepo into subgraph-refactor

0xdavinchee committed:
bytes id refactor

* use bytes id
* this will break the build but is primarily for seeing what the new id's look like

0xdavinchee committed:
Revert "bytes id refactor"

This reverts commit 762291b4ce733cd0e5401eb4cadb8a3c7de7ed56.

0xdavinchee committed:
Test a single bytes id

* changing all the id to `Bytes!` causes a weird error with graph node where it says the graph is still being indexed when I try to make my first query

0xdavinchee committed:
Revert "Test a single bytes id"

This reverts commit 545c469c6a40fb73fb3ad712635120ea52932d69.

0xdavinchee committed:
change query to get array of ATS to print

0xdavinchee committed:
keccak test

0xdavinchee committed:
bytearray to bytes

0xdavinchee committed:
ATS test fix

0xdavinchee committed:
validator cleanup

* remove id requirement from aggregate validator functions - rely on the expected entity id
* commit to test if the tests are passing

0xdavinchee committed:
proper ATS id all around!

0xdavinchee committed:
tokenStats ID is Bytes! type now

0xdavinchee committed:
more singular bytes id's

0xdavinchee committed:
no token bytes id might work?

is token bytes the issue?

0xdavinchee committed:
should pass tests

tests should pass

kasparkallas commented on packages/subgraph/schema.graphql:
```diff
@@ -1028,9 +1030,9 @@ interaction with a `token`.
+    ID composed of: accountID + keccak256("-") + tokenID (without the 0x)
     """
-    id: ID!
+    id: Bytes!
```
This does not look right... ? The field is `Bytes` but it's actually a concatenated string.

kasparkallas commented on packages/subgraph/schema.graphql:
```diff
@@ -1094,9 +1096,9 @@ TokenStatistic: An aggregate entity which aggregates data of a single `token`.
+    id: token address
     """
-    id: ID!
+    id: Bytes!
```
Same here. "id" should always be "ID", no?

kasparkallas commented on packages/subgraph/src/mappings/idav1.ts:
```diff
@@ -423,7 +418,7 @@ export function handleSubscriptionUnitsUpdated(
         return;
     }
-    let tokenId = event.params.token.toHex();
+    let tokenId = event.params.token;
```
Does not look like this is used?

0xdavinchee committed:
account entity fix

ensure account exists before trying to access property (probably what is breaking the indexing)

0xdavinchee committed:
Revert "account entity fix"

This reverts commit f9f62b8965f756118a0c7d9df567fdda49f11f54.

0xdavinchee committed:
merge dev

0xdavinchee committed:
Merge branch 'dev' into subgraph-refactor

0xdavinchee committed:
Merge branch 'dev' into subgraph-refactor

0xdavinchee committed:
cleanup

* remove instances of Bytes! id - use ID!
* remove .concat, use +
* remove unused consts/funcs since we are no longer using bytes id's
* remove consoles
* fix test case to accurately represent comment

0xdavinchee committed:
refactor cleanup

* renaming of propId => propAddress where relevant
* getID always takes the raw address and handles the conversion to string within the function
* Bytes types converted to Address as this is more accurate (what we usually get from the events)
* always save in init case

0xdavinchee committed:
build fix

0xdavinchee committed:
further cleanup

* remove unused
* check proper zero address not new Address(0)

0xdavinchee committed:
no 0 address ATS

0xdavinchee committed:
is this the breaking change?

0xdavinchee committed:
ats filter out

0xdavinchee committed:
tests should pass now

0xdavinchee committed:
build fix

0xdavinchee committed:
maybe the object isn't null

0xdavinchee committed:
should pass

0xdavinchee committed:
incorrect logic fixed up

0xdavinchee committed:
test w/ hardhat

0xdavinchee committed:
build ci fix

0xdavinchee committed:
missing -

0xdavinchee committed:
revert a commit

0xdavinchee committed:
revert commits

0xdavinchee committed:
proper zero address handling

hellwolf commented:
@msoni89 please also have a look

msoni89 commented on packages/subgraph/src/mappingHelpers.ts:
```diff
@@ -82,7 +78,7 @@ export function getOrInitSuperToken(
     let resolverAddress = getResolverAddress();
 
-    if (tokenId == ZERO_ADDRESS.toHex()) {
+    if (tokenAddress.equals(ZERO_ADDRESS)) {
```
Please early return it, is possible

0xdavinchee replied:
done

0xdavinchee committed:
remove unnecessary tokenId variable

0xdavinchee committed:
Merge branch 'dev' into subgraph-refactor

0xdavinchee committed:
early return

0xdavinchee committed:
Merge branch 'dev' into subgraph-refactor

hellwolf commented:
@kasparkallas @msoni89 please revisit

0xdavinchee committed:
Merge branch 'dev' into subgraph-refactor

hellwolf committed:
Merge branch 'dev' into subgraph-refactor
